wasm.md

```diff
@@ -1514,15 +1506,12 @@ Then, the surrounding `module` tag is discarded, and the definitions are execute
 
 ```k
     syntax Stmt       ::= ModuleDecl
-    syntax ModuleDecl ::= "(" "module" Defns ")"
-                        | "(" "module" Identifier Defns ")"
+    syntax ModuleDecl ::= "(" "module" OptionalId Defns ")"
                         |     "module" Map
  // --------------------------------------
-    rule <k> ( module ID:Identifier DEFNS ) => ( module DEFNS ) ... </k>
-         <moduleIds> ... .Map => ID |-> NEXT ... </moduleIds>
+    rule <k> ( module ID:OptionalId DEFNS ) => module structureModule(DEFNS) ... </k>
          <nextModuleIdx> NEXT </nextModuleIdx>
-
-    rule <k> ( module DEFNS ) => module structureModule(DEFNS) ... </k>
+         <moduleIds> IDS => #saveId(IDS, ID, NEXT) </moduleIds>
```

Comment:
I still don't like how we're reading `<nextModuleIdx>` but not incrementing it, since it's always supposed to be fresh.

Can we instead put these two cells `<nextModuleIdx>` and `<moduleIds>` in the next rule for `module MOD:Map`? Then `<nextModuleIdx>` is accessed and updated simultaneously, which is much more appropriate for a fresh-constants cell.

Reply:
That's fair, and in this case quite easy, too. I'm not sure it will always be easy, for example when I'm desugaring all variants of imports/exports. But I'll have a closer look.

Reply:
Fixed.